controllers/workspace/metrics/deployment_provisioning.go

```diff
@@ -37,15 +35,14 @@ var infrastructureFailures = []string{
 // DetermineProvisioningFailureReason scans a deployment provisioning status info message
 // and returns the corresponding failure reason.
 // If a failure reason cannot be found, an Unknown reason is returned.
-func DetermineProvisioningFailureReason(status workspace.DeploymentProvisioningStatus) FailureReason {
-	failStartupMsg := status.Info()
+func DetermineProvisioningFailureReason(statusMessage string) FailureReason {
 	for _, failure := range badRequestFailures {
-		if strings.Contains(failStartupMsg, failure) {
+		if strings.Contains(statusMessage, failure) {
```

Comment:
I called the function parameter `statusMessage` to make it more inline with the function documentation, but maybe I should have just kept `failStartupMsg`. Was also considering `statusInfoMessage` or `deploymentInfoMessage`